src/app/shared/components/components/add-product-modal/add-product-modal.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, Output, EventEmitter } from '@angular/core';
+import {
+  FormBuilder,
+  Validators,
+} from '@angular/forms';
+import { ProductsService } from 'src/app/core/products.service';
+import { Product } from '../../../../product.interface';
+
+@Component({
+  selector: 'app-add-product-modal',
+  templateUrl: './add-product-modal.component.html',
+  styleUrls: ['./add-product-modal.component.scss'],
+})
+export class AddProductModalComponent {
+
+  private numberRegEx = /^\s*(?=.*[1-9])\d*(?:\.\d{1,2})?\s*$/;
+
+  addProductForm = this.fb.group({
+    name: ['', Validators.required],
+    photoUrl: [''],
+    price: [
+      0,
+      Validators.compose([
+        Validators.pattern(this.numberRegEx),
+        Validators.required,
+        Validators.min(0.01),
+      ]),
+    ],
+    productDescription: [''],
+    flagged: [false],
+  });
+
+  @Output() onCloseAddProductModalEvent = new EventEmitter();
+  @Output() onAddProductEvent = new EventEmitter<Product>();
```

Comment:
As a convention event emitters should not have a prefix `on`, nor a suffix event. Example: `closeModal`, `addProduct`